backend/src/main/java/gov/cdc/usds/simplereport/service/LoincService.java

```diff
@@ -0,0 +1,311 @@
+            .filter(component -> Objects.equals(component.getName(), "property"))
+            .toList();
+    for (Parameters.ParametersParameterComponent property : properties) {
+      String firstValue = property.getPart().get(0).getValue().toString();
+      Type lastValue = property.getPart().get(1).getValue();
+
+      switch (firstValue) {
+        case "category":
+          {
+            Coding coding = (Coding) lastValue;
+            if (coding.getCode().equals("LP29693-6")) {
+              isLab = true;
+            }
+            break;
+          }
+        case "DefinitionDescription":
+          {
+            description = lastValue.toString();
+            break;
+          }
+        case "LONG_COMMON_NAME":
+          {
+            longCommonName = lastValue.toString();
+            break;
+          }
+        case "SCALE_TYP":
+          {
+            Coding coding = (Coding) lastValue;
+            scaleCode = coding.getCode();
+            scaleDisplay = coding.getDisplay();
+            break;
+          }
+        case "system-core":
+          {
+            Coding coding = (Coding) lastValue;
+            systemCode = coding.getCode();
+            systemDisplay = coding.getDisplay();
+            break;
+          }
+        case "answer-list":
+          {
+            answerList = lastValue.toString();
+            break;
+          }
+        case "ORDER_OBS":
+          {
+            orderOrObservation = lastValue.toString();
+            break;
+          }
+        case "PanelType":
+          {
+            if (lastValue.toString().equals("Panel")) {
+              panel = true;
+            }
+            break;
+          }
+      }
+    }
+
+    if (!isLab) {
+      return Optional.empty();
+    }
+
+    return Optional.of(
+        new Lab(
+            loinc.getCode(),
+            display,
+            description,
+            longCommonName,
+            scaleCode,
+            scaleDisplay,
+            systemCode,
+            systemDisplay,
+            answerList,
+            orderOrObservation,
+            panel));
+  }
+
+  // TODO write a unit test
+  private List<Lab> reduceLabs(List<Lab> labs, List<LoincStaging> loincs) {
+    List<String> codes = new ArrayList<>();
+    List<Lab> labsToSave = new ArrayList<>();
+
+    for (int i = 0; i < labs.size(); i++) {
+
+      String code = labs.get(i).getCode();
+      if (codes.contains(code)) {
+        continue;
+      }
+      codes.add(code);
+
+      Lab lab = labs.get(i);
+      Optional<Lab> foundLab = labRepository.findByCode(lab.getCode());
+      if (foundLab.isPresent()) {
+        lab = foundLab.get();
+      }
+
+      for (int j = 0; j < labs.size(); j++) {
+        if (code.equals(labs.get(j).getCode())) {
+          Condition conditionToAdd = loincs.get(j).getCondition();
+          if (!lab.getConditions().contains(conditionToAdd)) {
+            lab.addCondition(conditionToAdd);
+          }
+        }
+      }
```

Comment:
could we do something like this here?
```
      for (LoincStaging loinc : loincs) {
        if (loinc.getCode().equals(code)) {
          Condition conditionToAdd = loinc.getCondition();
          if (!lab.getConditions().contains(conditionToAdd)) {
            lab.addCondition(conditionToAdd);
          }
        }
      }
```

do we need to maintain a set of parallel arrays or a map of code to lab? very possible I'm misunderstanding but the code in LoincStaging object should correspond directly to the code of a Lab object right? the only thing we need from the `loincs` list is the map of a loinc code to a condition